src/spec/unit.ts

```diff
@@ -39,18 +38,39 @@ export interface GenericUnitSpec<E extends Encoding<any>, M> extends BaseSpec, L
 /**
  * A unit specification without any shortcut/expansion syntax.
  */
-export type NormalizedUnitSpec = GenericUnitSpec<Encoding<string | RepeatRef>, Mark | MarkDef>;
+export type NormalizedUnitSpec<
+  /** Extra Encoding */
+  EE = {}
+> = GenericUnitSpec<Encoding<Field> & EE, Mark | MarkDef>;
+
+/* tslint:disable */
+// Need to declare empty object so the generated schema has a reasonable name for ExtendedUnitSpec
+export interface EmptyObject {}
+/* tslint:enable */
 
 /**
- * Unit spec that can have a composite mark.
+ * Unit spec that can be normalized/expanded into a layer spec or another unit spec.
  */
-export type CompositeUnitSpec = GenericUnitSpec<Encoding<string | RepeatRef>, AnyMark>;
+export type ExtendedUnitSpec<
+  /** Extra Encoding */
+  EE = EmptyObject
+> = NormalizedUnitSpec<EE> | CompositeMarkUnitSpec<EE>;
 
 /**
- * Unit spec that can have a composite mark and row or column channels.
+ * Unit spec that can have a composite mark and row or column channels (shorthand for a facet spec).
  */
-export type FacetedCompositeUnitSpec = GenericUnitSpec<EncodingWithFacet<string | RepeatRef>, AnyMark>;
+export type FacetedExtendedUnitSpec = ExtendedUnitSpec<FacetMapping<Field>>;
+
+// Note: The following three lines are equivalent to:
```

Comment:
@domoritz I found a workaround, so you don't have to rush to fix this.

Reply:
Great. I should still look into it to make sure it’s not a problem elsewhere.